docker.go

```diff
@@ -634,7 +634,14 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 
 	c.stopProducer = make(chan bool)
 
+	ctx, cancel := context.WithCancel(ctx)
+
 	go func(stop <-chan bool) {
+		<-stop
```

Comment:
Can you add a little context in this goroutine? IIUC we want to make sure the stopProducer channel's content is read, but I may be wrong. Is this assumption correct?

Reply:
Actually, this is to cancel the context when `c.StopLogProducer()` is called. I've removed the goroutine altogether and added a call to `cancel` directly to `c.StopLogProducer`.

Reply:
By the way, should `c.stopProducer` be protected by a mutex? It wasn't protected before so I didn't add the mutex, but I don't see any documentation saying whether the methods can be called concurrently or not.